sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/ObjectExpressionsSuite.scala

```diff
@@ -66,4 +66,16 @@ class ObjectExpressionsSuite extends SparkFunSuite with ExpressionEvalHelper {
     checkEvalutionWithUnsafeProjection(
       mapEncoder.serializer.head, mapExpected, mapInputRow)
   }
+
+  test("SPARK-23586: UnwrapOption should support interpreted execution") {
```

Comment:
I just found out that there are literally no unit tests for this expression :(...

Can you use `checkEvaluation`? This will make sure both code gen and interpreted mode are tested?